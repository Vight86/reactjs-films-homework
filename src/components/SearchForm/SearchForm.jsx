import React, { useState } from 'react';
import propTypes from 'prop-types';
import styles from './SearchForm.scss';

const SearchForm = ({ handleSearchSubmit }) => {
  const [value, setValue] = useState('');

  return (
    <form
      className={styles.searchForm}
      onSubmit={(e) => {
        handleSearchSubmit(value, e);
        setValue('');
      }}
    >
      <input
        value={value}
        onChange={(e) => setValue(e.target.value)}
        className={styles.searchForm__input}
      />
      <button
        type="submit"
        aria-label="Search"
        className={styles.searchForm__button}
      />
    </form>
  );
};

SearchForm.propTypes = {
  handleSearchSubmit: propTypes.func.isRequired,
};

export default SearchForm;
