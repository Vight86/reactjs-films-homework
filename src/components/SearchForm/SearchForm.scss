@import "../../utils/_variables.scss";

.searchForm {
  position: relative;
  width: 50%;
  max-width: 53em;
  height: 6.5em;
  box-sizing: border-box;

  &__input {
    width: 100%;
    height: 100%;
    box-sizing: border-box;
    padding-right: 2.8em;
    padding-left: 0.9em;
    border: 3px solid currentColor;
    background: transparent;
    border-radius: 0.2em;
    box-shadow: 0 0 0 currentColor;
    color: $color-light;
    font-family: $font-primary;
    font-size: 2.8em;
    line-height: 6.5em;
    transition: box-shadow 0.3s;

    &:focus {
      box-shadow: 0 0 20px currentColor;
      outline: none;
    }
  }

  &__button {
    position: absolute;
    top: 0;
    right: 0;
    width: 4.9em;
    height: 4.9em;
    box-sizing: border-box;
    border: none;
    background: transparent;
    background-image: url('./search-icon.svg');
    background-position: center;
    background-repeat: no-repeat;
    outline: none;

    &:hover,
    &:focus {
      background-size: 60%;
      cursor: pointer;
    }
  }
}
