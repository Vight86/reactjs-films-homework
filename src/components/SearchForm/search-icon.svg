<svg xmlns="http://www.w3.org/2000/svg" width="28" height="28" viewBox="0 0 28 28">
  <defs>
    <style>
      .cls-1 {
        fill: #fff;
        fill-rule: evenodd;
      }
    </style>
  </defs>
  <path id="Search_Icon" data-name="Search Icon" class="cls-1" d="M2262,72.3l-8.26-8.269a11.088,11.088,0,1,0-1.69,1.7L2260.3,74Zm-23.35-8.758a9.091,9.091,0,1,1,6.43,2.661A9.034,9.034,0,0,1,2238.65,63.543Z" transform="translate(-2234 -46)"/>
</svg>
