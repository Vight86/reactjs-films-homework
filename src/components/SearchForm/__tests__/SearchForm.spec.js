import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import SearchForm from '../index';

let mockHandleSearchSubmit;
let renderer;
let result;

beforeEach(() => {
  mockHandleSearchSubmit = jest.fn();
  renderer = new ShallowRenderer();
  renderer.render(<SearchForm handleSearchSubmit={mockHandleSearchSubmit} />);
  result = renderer.getRenderOutput();
});

afterEach(() => {
  mockHandleSearchSubmit.mockClear();
  renderer = null;
  result = null;
});

describe('render SearchForm component', () => {
  it('render correctly', () => {
    expect(result).toMatchSnapshot();
  });

  it('handle onChange event correctly', () => {
    const [input] = result.props.children;
    const mockEvent = { target: { value: 'test' } };
    input.props.onChange(mockEvent);
    result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('handle submit event correctly', () => {
    result.props.onSubmit();
    result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
