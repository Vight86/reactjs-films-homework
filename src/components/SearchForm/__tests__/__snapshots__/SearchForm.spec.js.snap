// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SearchForm component handle onChange event correctly 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    value="test"
  />
  <button
    aria-label="Search"
    className="searchForm__button"
    type="submit"
  />
</form>
`;

exports[`render SearchForm component handle submit event correctly 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    value=""
  />
  <button
    aria-label="Search"
    className="searchForm__button"
    type="submit"
  />
</form>
`;

exports[`render SearchForm component render correctly 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    className="searchForm__input"
    onChange={[Function]}
    value=""
  />
  <button
    aria-label="Search"
    className="searchForm__button"
    type="submit"
  />
</form>
`;
